compiler/src/iree/compiler/Dialect/Stream/Transforms/RefineUsage.cpp

```diff
@@ -312,6 +312,15 @@ struct ApplyStreamableOp : public UsageRefinementPattern<Op> {
         cast<IREE::Util::SizeAwareOpInterface>(op.getOperation());
     for (unsigned i = 0; i < op->getNumResults(); ++i) {
       auto result = op->getResult(i);
+      // If it is an alloca op with 0 as input, it will cause an infinite loop
+      // were it fetches itself to allocate 0 bytes. Thus, don't handle this
+      // case. Happens when there is a zero-size tensor.
+      if (isa<IREE::Stream::AsyncAllocaOp>(op)) {
+        Value defOp = cast<IREE::Stream::AsyncAllocaOp>(op).getResultSize(0);
+        if (defOp.getDefiningOp() &&
+            cast<arith::ConstantIndexOp>(defOp.getDefiningOp()).value() == 0)
+          continue;
+      }
```

Comment:
I don't understand this - can you describe why this is specific to an alloca and zero length? Doing this specific op check and constant check in here has a strong smell to it and I'd like to understand what the cause is before we hack this.

Reply:
I mentioned this in the bug, basically what is happening is that in here (https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Dialect/Stream/Transforms/RefineUsage.cpp#L314) result is fetching the next instruction and then it is stuck on the same instruction and looping infinitely... So I am getting outputs like this 

```
%1 = stream.async.transfer %0 : !stream.resource<constant>{%c0} -> !stream.resource<transient>{%c0}
  %2 = stream.async.transfer %1 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %3 = stream.async.transfer %2 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %4 = stream.async.transfer %3 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %5 = stream.async.transfer %4 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %6 = stream.async.transfer %5 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %7 = stream.async.transfer %6 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
  %8 = stream.async.transfer %7 : !stream.resource<transient>{%c0} -> !stream.resource<transient>{%c0}
...
```

My expertise in this part of code (and all of IREE) is limited so I could be wrong here. One thing I noticed is that when the tensor is non-zero, it is breaking it into splats but when it is zero it is breaking into alloca expression. I have inputs to each of the passes, which is a bit long to paste here but I can send it to you by email if you wish.

Another question I do have is, what does 0-length tensor mean? Is it a dead-code that the compiler can remove it or should it mean something else?